setup/main.go

```diff
@@ -0,0 +1,1653 @@
+		// N.B.: there's no way to tell whether the collector actually needs to reload; we
+		// force this once per setup helper invocation (this starts out as false and is set
+		// to true below)
+		return state.DidReload, nil
+	},
+	Run: func(state *SetupState) error {
+		var doReload bool
+		err := survey.AskOne(&survey.Confirm{
+			Message: "The collector configuration must be reloaded for changes to take effect; reload now?",
+			Default: false,
+		}, &doReload)
+		if err != nil {
+			return err
+		}
+		if !doReload {
+			return nil
+		}
+		err = reload()
+		if err != nil {
+			return err
+		}
+		state.DidReload = true
+		return nil
+	},
+}
+
+var restartPg = &Step{
+	Description: "If necessary, restart Postgres to have configuration changes take effect",
+	Check: func(state *SetupState) (bool, error) {
+		row, err := state.QueryRunner.QueryRow("SELECT COUNT(*) FROM pg_settings WHERE pending_restart;")
+		if err != nil {
+			return false, err
+		}
+		return row.GetInt(0) == 0, nil
+	},
+	Run: func(state *SetupState) error {
+		rows, err := state.QueryRunner.Query("SELECT name FROM pg_settings WHERE pending_restart")
+		if err != nil {
+			return err
+		}
+		var pendingSettings []string
+		for _, row := range rows {
+			pendingSettings = append(pendingSettings, row.GetString(0))
+		}
+
+		pendingList := getConjuctionList(pendingSettings)
+		var restartNow bool
+		err = survey.AskOne(&survey.Confirm{
+			Message: fmt.Sprintf("Postgres must be restarted for changes to %s to take effect; restart Postgres now?", pendingList),
+			Default: false,
+		}, &restartNow)
+		if err != nil {
+			return err
+		}
+
+		if !restartNow {
+			return nil
+		}
+
+		return service.RestartPostgres()
+	},
+}
+
+// N.B.: this needs to happen *after* the Postgres restart so that ALTER SYSTEM
+// recognizes these as valid configuration settings
+var configureAutoExplain = &Step{
+	Description: "Review auto_explain settings",
+	Check: func(state *SetupState) (bool, error) {
+		if state.SkipAutomatedExplain {
+			return true, nil
+		}
+		logExplain, err := usingLogExplain(state.CurrentSection)
+		if err != nil || logExplain {
+			return logExplain, err
+		}
+		if state.SkipAutoExplainRecommendedSettings {
+			return true, nil
+		}
+
+		return false, nil
+	},
+	Run: func(state *SetupState) error {
+		var doReview bool
+		err := survey.AskOne(&survey.Confirm{
+			Message: "Review auto_explain configuration settings?",
+			Default: false,
+			Help:    "Optional, but will ensure best balance of monitoring visibility and performance; review these settings at https://www.postgresql.org/docs/current/auto-explain.html#id-1.11.7.13.5",
+		}, &doReview)
+		if err != nil {
+			return err
+		}
+		if !doReview {
+			state.SkipAutoExplainRecommendedSettings = true
+			return nil
+		}
+
+		var logAnalyzeIdx int
+		var logAnalyzeOpts = []string{"on", "off"}
+		err = survey.AskOne(&survey.Select{
+			Message: "Set auto_explain.log_analyze to (will be saved to Postgres):",
```

Comment:
Should we maybe just ask `Use recommended auto_explain configuration?` instead of checking each setting individually?

Reply:
Maybe? I wanted to be really clear about what we're doing at every step. I suppose we could show you all the settings we would change, but if you want to change some of them but not all, that's frustrating. We should probably, though, only change settings that are not already set to the recommended values, right?

Reply:
Fair point - yes, it would be good if we limited it to the settings that are not configured correctly yet